cmd/defenseclaw-setup/main.go

```
@@ -854,28 +837,58 @@ func validateInstall(root, version string) error {
+		return errors.New("machine-readable version contains trailing content")
+	}
+	if report.SchemaVersion != 1 || report.Name != expectedName {
+		return fmt.Errorf("unexpected version identity schema=%d name=%q", report.SchemaVersion, report.Name)
+	}
+	if !validPayloadVersion(report.Version) || report.Version != expectedVersion {
+		return fmt.Errorf("reported version %q does not exactly match packaged version %q", report.Version, expectedVersion)
 	}
 	return nil
 }
 
 func runInitialConfiguration(root, dataRoot string, opts options) error {
```

golangci-lint (2.12.2)

[error] func runInitialConfiguration is unused

(unused)